### PR TITLE
Custom Admin menu item edit: Display Title, Parent Item and Ordering translations

### PR DESCRIPTION
**Pull request by infograf768**
Following up on https://github.com/joomla/joomla-cms/pull/19900#issuecomment-373315563

### Summary of Changes
Adds a readonly fake field to display the translation of the Title when that one has been saved as a MOD_MENU_ variant after using the presets.
Translating the menu items concerned in the Ordering and Parent Item dropdowns

### Testing Instructions
Same as  https://github.com/joomla/joomla-cms/pull/19900
Go to Menu Manager
Add a new Menu, type administrator, select one of the options predefined Joomla / predefined modern
Store the new menu.

Open the menu Manager and edit one of the menu items of this custom admin menu.
### Before patch
Only the language constants are displayed.

### After patch
screen shot 2018-03-15 at 10 34 40

screen-shot-title-translation

@Quy @chmst @izharaazmi

### CODE REVIEW

File: administrator/language/en-GB/en-GB.com_menus.ini [hunk 1]
```diff
@@ -213,6 +213,7 @@ COM_MENUS_SUCCESS_REORDERED="Menu item reordered."
 COM_MENUS_TIP_ALIAS_LABEL="<strong>Warning!</strong><br />Leave the alias field empty if the menu item alias and the menu item linked to by the alias have the same parent."
 COM_MENUS_TIP_ASSOCIATION="Associated menu items"
 COM_MENUS_TITLE_EDIT_ITEM="Menu Manager: Title Edit Item"
+COM_MENUS_TITLE_TRANSLATION="Title Translation"
```
Comment (izharaazmi) [inline administrator/language/en-GB/en-GB.com_menus.ini]: May be `Title (en-GB)` where language code is current language should be better?

File: administrator/components/com_menus/views/item/tmpl/edit.php [hunk 2]
```diff
@@ -114,11 +114,29 @@
...
+		<?php if ($clientId === 1) : ?>
+			<div class="control-group form-inline-header">
+				<div class="form-inline-header control-label">
+					<label><?php echo JText::_('COM_MENUS_TITLE_TRANSLATION'); ?></label>
+				</div>
+				<div class="form-inline-header controls">
+					<input class="input-xlarge" value="<?php echo JText::_($this->item->title); ?>" readonly="" type="text">
+				</div>
+			</div>
+		<?php endif; ?>
```
Comment (izharaazmi) [inline administrator/components/com_menus/views/item/tmpl/edit.php]: Isn't this block extra-indented?

File: administrator/language/en-GB/en-GB.com_menus.ini (hunk 1 above)
Comment (infograf768) [reply administrator/language/en-GB/en-GB.com_menus.ini]: Let me see this

File: administrator/components/com_menus/views/item/tmpl/edit.php (hunk 2 above)
Comment (infograf768) [reply administrator/components/com_menus/views/item/tmpl/edit.php]: No, it is fine

Comment (infograf768) [reply administrator/components/com_menus/views/item/tmpl/edit.php]: I see now what you mean. The whole block indeed. Will change.

**Comment by infograf768** [comment]
@izharaazmi 
done. Nicer indeed. Here when using French:

screen shot 2018-03-15 at 16 13 11

Note: Title is in English above as we have no French string yet.

File: administrator/components/com_menus/views/item/tmpl/edit.php [hunk 3]
```diff
@@ -114,13 +114,31 @@
...
+	JFactory::getLanguage()->load('mod_menu', JPATH_ADMINISTRATOR, null, false, true);
+}
 ?>
 <form action="<?php echo JRoute::_('index.php?option=com_menus&view=item&client_id=' . $clientId . '&layout=' . $layout . $tmpl . '&id=' . (int) $this->item->id); ?>" method="post" name="adminForm" id="item-form" class="form-validate">
 
 	<?php echo JLayoutHelper::render('joomla.edit.title_alias', $this); ?>
 
-	<div class="form-horizontal">
+	<?php // Add the translation of the menu item title when client is administrator ?>
+	<?php if ($clientId === 1) : ?>
```
Comment (Quy) [inline administrator/components/com_menus/views/item/tmpl/edit.php]: Above `==` and here `===`. Which one?

File: administrator/components/com_menus/views/item/tmpl/edit.php [hunk 4]
```diff
@@ -114,13 +114,31 @@
...
+		<div class="control-group form-inline-header">
```
Comment (Quy) [inline administrator/components/com_menus/views/item/tmpl/edit.php]: Can this section be in the xml file and use `$this->form->renderField`?

File: administrator/components/com_menus/views/item/tmpl/edit.php (hunk 3 above)
Comment (infograf768) [reply administrator/components/com_menus/views/item/tmpl/edit.php]: corrected

File: administrator/components/com_menus/views/item/tmpl/edit.php (hunk 4 above)
Comment (infograf768) [reply administrator/components/com_menus/views/item/tmpl/edit.php]: I did not find a way to get $this->item->title though a field in the xml

**Comment by izharaazmi** [comment]
If we really want to go into the detail, I would recommend this to go into `joomla.edit.title_alias` layout and wherever the title is all UPPERCASE and comprises of valid characters for a joomla language key, then we show this.

File: administrator/components/com_menus/views/item/tmpl/edit.php [hunk 5]
```diff
@@ -114,13 +114,31 @@
 $layout   = $isModal ? 'modal' : 'edit';
 $tmpl     = $isModal || $input->get('tmpl', '', 'cmd') === 'component' ? '&tmpl=component' : '';
 $clientId = $this->state->get('item.client_id', 0);
+$lang     = JFactory::getLanguage()->getTag();
```
Comment (Quy) [inline administrator/components/com_menus/views/item/tmpl/edit.php:117]: Maybe move this after line 121 since that is when you only need it?

Comment (izharaazmi) [reply administrator/components/com_menus/views/item/tmpl/edit.php:117]: Well its needed twice, used once though. Line 122 can be refactored.

Comment (infograf768) [reply administrator/components/com_menus/views/item/tmpl/edit.php:117]: it is imho always better to fetch that type of general variables in top of file.

Comment (izharaazmi) [reply administrator/components/com_menus/views/item/tmpl/edit.php:117]: I agree.

**Comment by brianteeman** [comment]
> If we really want to go into the detail, I would recommend this to go into joomla.edit.title_alias layout and wherever the title is all UPPERCASE and comprises of valid characters for a joomla language key, then we show this.

That's no good and is similar to why I had to create #19802 

If you have a menu item in capitals that matches a key it will be translated even if its just a conincidence

**Comment by infograf768** [comment]
> If we really want to go into the detail, I would recommend this to go into joomla.edit.title_alias layout and wherever the title is all UPPERCASE and comprises of valid characters for a joomla language key, then we show this.

The issue is only present in this special case i.e. for custom admin menus. Would'nt that be overkill?

**Comment by izharaazmi** [comment]
Yes, my bad.

**Comment by izharaazmi** [comment]
Shouldn't we hide this for New menu items?

**Comment by infograf768** [comment]
> Shouldn't we hide this for New menu items?

You mean by changing line 130 to something like
`<?php if ($clientId === 1 && !empty($this->item->id)) : ?>`
(Maybe !empty is not even necessary)

**Comment by izharaazmi** [comment]
Yes `$this->item->id == 0` should be fine.

**Comment by infograf768** [comment]
Done

**Comment by Quy** [comment]
The vertical alignment of the Title label is lower than the value in the input field. See `Link` field to compare. The markup is not quite right.

**Comment by izharaazmi** [comment]
Now the only issue that stays, is that we are limited to the language keys in mod_menu, com_menus. If we need to and can, we should think for a solution. Loading all languages are certainly NOT a solution 😄

**Comment by izharaazmi** [comment]
I have tested this item :white_check_mark: successfully on b1eb953df8f1ea5e756ed605d01d29a460659369<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19916">issues.joomla.org/tracker/joomla-cms/19916</a>.</sub>

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on b1eb953df8f1ea5e756ed605d01d29a460659369<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19916">issues.joomla.org/tracker/joomla-cms/19916</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

**Comment by infograf768** [comment]
@Quy 
Corrected label alignment.
screen shot 2018-03-16 at 09 04 55

No need to test again. Let's keep it RTC.

File: administrator/components/com_menus/views/item/tmpl/edit.php [hunk 6]
```diff
@@ -114,13 +114,31 @@
...
+}
 ?>
 <form action="<?php echo JRoute::_('index.php?option=com_menus&view=item&client_id=' . $clientId . '&layout=' . $layout . $tmpl . '&id=' . (int) $this->item->id); ?>" method="post" name="adminForm" id="item-form" class="form-validate">
 
 	<?php echo JLayoutHelper::render('joomla.edit.title_alias', $this); ?>
 
-	<div class="form-horizontal">
+	<?php // Add the translation of the menu item title when client is administrator ?>
+	<?php if ($clientId === 1 && $this->item->id != 0) : ?>
+		<div class="form-inline form-inline-header control-group">
```
Comment (Quy) [inline administrator/components/com_menus/views/item/tmpl/edit.php]: Please move `control-group` to its own `<div>` as follows to give more space between the field and the tabs.
```
		<div class="form-inline form-inline-header">
                    <div class="control-group">
			<div class="control-label">
				<label><?php echo JText::sprintf('COM_MENUS_TITLE_TRANSLATION', $lang); ?></label>
			</div>
			<div class="controls">
				<input class="input-xlarge" value="<?php echo JText::_($this->item->title); ?>" readonly="" type="text">
			</div>
                    </div>
		</div>
```

File: administrator/components/com_menus/views/item/tmpl/edit.php [hunk 7]
```diff
@@ -114,13 +114,31 @@
...
+			<div class="control-label">
+				<label><?php echo JText::sprintf('COM_MENUS_TITLE_TRANSLATION', $lang); ?></label>
+			</div>
+			<div class="controls">
+				<input class="input-xlarge" value="<?php echo JText::_($this->item->title); ?>" readonly="" type="text">
```
Comment (Quy) [inline administrator/components/com_menus/views/item/tmpl/edit.php]: Change to `readonly="readonly"`.

**Comment by infograf768** [comment]
@Quy Cosmetic changes done.

**Comment by infograf768** [comment]
why to 3.9.0 ?

**Comment by infograf768** [comment]
@mbabker 
although this contains a new string, this should go imho into 3.8.7 as it is the logical suite of https://github.com/joomla/joomla-cms/pull/19900

**Comment by brianteeman** [comment]
its a new feature not a bug fix

**Comment by infograf768** [comment]
It is not really a hew feature. It helps users who are lost with these constants

**Comment by infograf768** [comment]
Thanks

**Comment by mbabker** [comment]
Although technically it is a feature, we've already merged minor feature additions to 3.8 that are bigger in scope than this (session metadata manager, update site mirrors).  So, I'm OK with this one.

**Comment by brianteeman** [comment]
and they shouldnt have been

**Comment by mbabker** [comment]
Not exactly in disagreement there, but the other option is every other release turns into a minor and we've already seen what kind of problems it causes when you add more minor releases and push merged things for a given minor branch back to another release.  Sometimes you just have to make a gut call on what's best for everyone.

**Pull request merged**